service/ProductManager.java

```diff
@@ -0,0 +1,66 @@
+package service;
+
+import exceptions.DuplicateProductException;
+import exceptions.ProductNotFoundException;
+import model.Product;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ProductManager {
+    private List<Product> inventory;
+
+    public ProductManager() {
+        inventory = new ArrayList<>();
+    }
+
+    public void addProduct(Product product) throws DuplicateProductException {
+        for (Product p : inventory) {
+            if (p.getId() == product.getId()) {
+                throw new DuplicateProductException("Product with ID " + product.getId() + " already exists.");
+            }
+        }
+        inventory.add(product);
+    }
+
+    public void updateProduct(Product product) throws ProductNotFoundException {
+        boolean found = false;
+        for (int i = 0; i < inventory.size(); i++) {
+            if (inventory.get(i).getId() == product.getId()) {
+                inventory.set(i, product);
+                found = true;
+                break;
+            }
+        }
+        if (!found) {
+            throw new ProductNotFoundException("Product with ID " + product.getId() + " not found.");
+        }
+    }
+
+    public void deleteProduct(int productId) throws ProductNotFoundException {
+        boolean found = false;
+        for (int i = 0; i < inventory.size(); i++) {
+            if (inventory.get(i).getId() == productId) {
+                inventory.remove(i);
+                found = true;
+                break;
+            }
+        }
+        if (!found) {
+            throw new ProductNotFoundException("Product with ID " + productId + " not found.");
+        }
+    }
+
+    public void sellProduct(int productId, int quantity) throws ProductNotFoundException {
+        for (Product product : inventory) {
+            if (product.getId() == productId) {
+                if (product.getQuantity() >= quantity) {
+                    product.setQuantity(product.getQuantity() - quantity);
+                } else {
+                    throw new ProductNotFoundException("Insufficient quantity for product with ID " + productId);
```

Comment:
this should not be a **ProductNotFoundException**

Reply:
Done